### ISSUE TITLE
Zoom in operation error

### ISSUE THREAD

**Issue by akihitoiwai0912** [densoGSR/lf_test#3]
### Original (Before) 
![image](https://github.com/user-attachments/assets/5a24012c-1610-460c-ab59-0119e9c545a2)

### Zoom in (After)
![image](https://github.com/user-attachments/assets/990b99e5-2c6c-4f51-a6b0-3c377d886992)
- **Problem: actor b and c were shorten than original ones**

### Zoom in (After)
![image](https://github.com/user-attachments/assets/c8f6265c-d3cf-4b7b-9210-2a8177818696)
- **Problem: actor a disappeared**

**Comment by elgeeko1** [comment]
The observations here are expected behavior.

> Problem: actor b and c were shorten than original ones

The time window feature in Grafana isn't a "zoom in" feature, but rather it changes the time window for a database query, the query is run, and the results displayed. The smaller time window in the second image includes the state transitions from `IDLE` to `EXECUTING` for reactors b and c, but omits the corresponding completion state change from `EXECUTING` to `IDLE` which occurs after the end of the time window. The Grafana State Transition panel reflects an instantaneous change in state with no known duration as a narrow bar -- it will appear this narrow regardless of how zoomed in you go. You're not able to hover over this event to show a duration, because no duration has been calculated for time window.

What I can't explain is why reaction d still appears as a non-instantaneous duration, since its completion also falls outside of the time window. I've issued a bug report for this and will review.

> Problem: actor a disappeared

Similar to above, the time window selected in the zoomed in view omits the start of the execution, so reactor a is omitted from the state transition graph. This is expected behavior.

Reaction e shouldn't appear and is the same bug as above.

**Comment by akihitoiwai0912** [comment]
Thank you very much for your detailed explanation! I understand the mechanism. I didn't realize that Grafana's time window feature wasn't a zoom in feature. However, we believe that the Zoom-in feature of the Timing sequence is necessary as a user interface.

**Comment by elgeeko1** [comment]
Yes a zoom feature on execution timelines would be useful. We may need to consider a tool other than Grafana for this. Grafana is designed to display single database queries, and limits how many data points may be returned by a single query. We've captured this feature request in our product backlog.
